### DISCUSSION TITLE
采集到的日志没有 对应的 pod/namesapce / _container_name_ 这些元信息

### DISCUSSION THREAD

**Discussion by weinshun** [alibaba/ilogtail#535] [Help] [answered]
通过daemonset 方式部署  ilogtail，发现日志没有 对应的 pod/namesapce  / _container_name_ 这些元信息。
这些k8s相关的元信息是通过特殊配置实现的，还是自动支持的。我的配置如下：
  regex_normal_log.yaml: |
    enable: true
    inputs:
      - Type: file_log
        LogPath: /logtail_host/log/kubelet/pods/*/volumes/kubernetes.io~empty-dir/*
        FilePattern: *.log
    processors:
    flushers:
      - Type: flusher_kafka
        Brokers:
          - xx.xx.xx.xx:xx
        Topic: normal-xx
通过查看 kafka topic数据  normal-xx，只看到日志的原始内容。
{"Time":1669637136,"Contents":[{"Key":"__tag__:__path__","Value":"/logtail_host/log/kubelet/pods/ff3d6c3a-6d6b-11ed-b4b4-eeeeeeeeeeee/volumes/kubernetes.io~empty-dir/vol2/2022-11-27_0.log"},{"Key":"__tag__:__user_defined_id__","Value":"default"},{"Key":"content","Value":"2022-11-27 02:12:33.102 {trace: xxx} DEBUG[http-nio-8080-exec-10] StrategyFilter.doFilter(82) - step -2 : url = 119.POST.api.anc.v1.eventList.v2.queryPage 没有配置对象，不需要后续处理"}]}

**Comment by yyuuttaaoo** [comment, marked as answer]
参考一下这里，使用容器内路径和DockerFile: true对容器内文件进行采集
https://github.com/alibaba/ilogtail/blob/main/docs/cn/getting-started/k8s-daemonset-to-kafka.md#%E9%87%87%E9%9B%86%E5%AE%B9%E5%99%A8%E5%86%85%E7%9A%84%E6%96%87%E4%BB%B6

**Reply by weinshun** [reply]
感谢回复哈。 要使用容器内路径，我配置的是 挂载到主机的 /logtail_host下的路径，经测试已经可以用了，感谢感谢~！
